tgui/packages/tgui/interfaces/CrewConsole.jsx

```diff
@@ -86,25 +87,119 @@ export const CrewConsole = () => {
 
 const CrewTable = (props) => {
   const { act, data } = useBackend();
-  const sensors = sortBy(data.sensors ?? [], (s) => s.ijob);
+  const sensors = data.sensors ?? [];
+
+  const [shownSensors, setShownSensors] = useState(sensors.slice());
+  const [sortColumn, setSortColumn] = useState('ijob');
+  const [sortAsc, setSortAsc] = useState(true);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const updateSortMode = (column) => {
+    if (column === sortColumn) {
+      if (!sortAsc) {
+        setSortColumn('ijob');
+        setSortAsc(true);
+      } else {
+        setSortAsc(false);
+      }
+    } else {
+      setSortColumn(column);
+      setSortAsc(true);
+    }
+  };
+
+  const healthSort = (a, b) => {
+    if (a.life_status < b.life_status) return -1;
+    if (a.life_status > b.life_status) return 1;
+    if (a.health > b.health) return -1;
+    if (a.health < b.health) return 1;
+    return 0;
+  };
+
+  useEffect(() => {
+    let unsorted = sensors.slice();
+    if (sortColumn === 'ijob') {
+      let sorted = sortBy(data.sensors ?? [], (s) => s.ijob).filter((crew) =>
+        crew.name.toLowerCase().includes(searchQuery.toLowerCase()),
+      );
+      setShownSensors(sorted);
+      return;
+    } else if (sortColumn === 'health') {
+      unsorted.sort(healthSort);
+    } else {
+      unsorted.sort((a, b) => {
+        a[sortColumn] ??= '~';
+        b[sortColumn] ??= '~';
+        if (a[sortColumn] < b[sortColumn]) return -1;
+        if (a[sortColumn] > b[sortColumn]) return 1;
+        return 0;
+      });
+    }
+
+    let sorted = unsorted.filter((crew) =>
+      crew.name.toLowerCase().includes(searchQuery.toLowerCase()),
+    );
+    if (!sortAsc) {
+      sorted.reverse();
+    }
+    setShownSensors(sorted);
+  }, [searchQuery, sensors, sortColumn, sortAsc]);
+
   return (
     <Table>
       <Table.Row>
-        <Table.Cell bold>Name</Table.Cell>
+        <Table.Cell bold>
+          <Button onClick={() => updateSortMode('name')}>
+            Name
+            {sortColumn === 'name' ? (
+              <Icon
+                style={{ marginLeft: '2px' }}
+                name={sortAsc ? 'chevron-up' : 'chevron-down'}
+              />
+            ) : (
+              <Icon style={{ marginLeft: '2px' }} name="minus" />
+            )}
+          </Button>
+          <Input
+            type="text"
+            placeholder="Search for name..."
+            onInput={(e) => setSearchQuery(e.target.value)}
+          />
```

Comment:
make this a section title or section buttons please, make the section scrollable, put the table in it

Reply:
Only problem with taking the buttons out of the table is that they won't line up with their respective columns